pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symboltable/VariableNameDeclaration.java

```diff
@@ -97,7 +98,8 @@ private TypeNode getTypeNode() {
         if (isPrimitiveType()) {
             return (TypeNode) getAccessNodeParent().getFirstChildOfType(ASTType.class).jjtGetChild(0);
         }
-        if (!isLambdaTypelessParameter()) {
+        if (!isLambdaTypelessParameter()
```

Comment:
On the latest master, this is replaced with the more future-proof `isTypeInferred` which relies on `VariableDeclaratorId.isTypeInferred`. If that method returned true for implicitly typed local vars, you could skip the next line's check

Reply:
I'll have a look. We should probably somehow combine the `isTypeInferred()` on `ASTVariableDeclaratorId` and on `ASTType`.